x-pack/platform/plugins/shared/embeddable_alerts_table/public/factories/alerts_table_embeddable_factory.tsx

```diff
@@ -42,69 +48,65 @@ export const getAlertsTableEmbeddableFactory = (
     const timeRange = initializeTimeRange(state);
     const titleManager = initializeTitleManager(state);
     const queryLoading$ = new BehaviorSubject<boolean | undefined>(true);
-    const { data, fieldFormats, licensing } = deps;
-    const { http, application, notifications, settings } = core;
+    const services = {
+      ...coreServices,
+      ...deps,
+    };
+    const tableConfig$ = new BehaviorSubject<EmbeddableAlertsTableConfig>(state.tableConfig);
     const api = buildApi(
       {
         ...timeRange.api,
         ...titleManager.api,
         dataLoading$: queryLoading$,
+        isEditingEnabled: () => true,
+        onEdit: async () => {
+          try {
+            const newTableConfig = await openConfigEditor({
+              coreServices,
+              parentApi: api.parentApi as PresentationContainer,
+              initialConfig: tableConfig$.getValue(),
+            });
+            tableConfig$.next(newTableConfig);
+          } catch {
+            // The user closed without saving
+          }
+        },
+        getTypeDisplayName: () => CONFIG_EDITOR_EDIT_TABLE_TITLE,
         serializeState: () => {
           return {
-            rawState: { ...titleManager.serialize(), ...timeRange.serialize() },
+            rawState: {
+              ...titleManager.serialize(),
+              ...timeRange.serialize(),
+              tableConfig: tableConfig$.getValue(),
+            },
           };
         },
       },
-      { ...titleManager.comparators, ...timeRange.comparators }
+      {
+        ...titleManager.comparators,
+        ...timeRange.comparators,
+        tableConfig: [tableConfig$, (value) => tableConfig$.next(value), (a, b) => deepEqual(a, b)],
+      }
     );
 
     return {
       api,
       Component: () => {
         const { timeRange: selectedTimeRange } = useFetchContext(api);
+        const [tableConfig] = useBatchedPublishingSubjects(tableConfig$);
```

Comment:
Nit: There's no reason to `useBatchedPublishingSubjects` if you only have one subject. You could try `useStateFromPublishingSubject` instead as `useBatchedPublishingSubjects` injects a `debounce(0)` which makes it slightly less responsive.

Reply:
Oh I didn't know this variant existed, switching to it, thanks! 🙂